### PR TITLE
Add Mypy type hint linter in Koalas lint check

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR adds Mypy type hint linter in our tests and Python linter process.

Closes https://github.com/databricks/koalas/issues/162

### CODE REVIEW

File: databricks/koalas/dask/utils.py [hunk 1]
```diff
@@ -1,56 +1,6 @@
 import functools
 from databricks.koalas.dask.compatibility import get_named_args
 
-_method_cache = {}
-
-
-class methodcaller(object):
```
Comment (HyukjinKwon) [inline databricks/koalas/dask/utils.py]: I think these are not used. mypy complained so I just removed.

File: databricks/__init__.py [hunk 2]
```diff
@@ -15,4 +15,4 @@
 #
 
 # https://packaging.python.org/guides/packaging-namespace-packages/#pkgutil-style-namespace-packages
-__path__ = __import__('pkgutil').extend_path(__path__, __name__)
+__path__ = __import__('pkgutil').extend_path(__path__, __name__)  # type: ignore
```
Comment (thunterdb) [inline databricks/__init__.py]: Yes, not much we can do here.

File: databricks/koalas/dask/utils.py (hunk 1 above)
Comment (thunterdb) [reply databricks/koalas/dask/utils.py]: That sounds great, thanks for the cleanup.

**Comment by thunterdb** [comment]
As discussed outside this thread:
 - strict mode is too strict for now
 - we should eventually add `spark-stubs` for more precise typing. This is to be done in subsequent work.

Merging the current work. Thanks!

**Pull request merged**